app/src/test/java/org/mozilla/fenix/crashes/CrashReporterControllerTest.kt

```diff
@@ -4,83 +4,114 @@
 
 package org.mozilla.fenix.crashes
 
-import androidx.navigation.NavController
-import androidx.navigation.NavDestination
 import io.mockk.every
 import io.mockk.mockk
+import io.mockk.spyk
 import io.mockk.verify
-import mozilla.components.lib.crash.Crash
+import mozilla.components.lib.crash.Crash.NativeCodeCrash
 import mozilla.components.support.test.ext.joinBlocking
-import org.junit.Before
 import org.junit.Test
-import org.mozilla.fenix.R
+import org.mozilla.fenix.components.AppStore
 import org.mozilla.fenix.components.Components
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.components.appstate.AppState
 
 import org.mozilla.fenix.utils.Settings
 
 class CrashReporterControllerTest {
 
-    private lateinit var components: Components
-    private lateinit var crash: Crash
-    private lateinit var sessionId: String
-    private lateinit var navContoller: NavController
-    private lateinit var settings: Settings
-
-    @Before
-    fun setup() {
-        components = mockk(relaxed = true)
-        crash = mockk()
-        sessionId = "testId"
-        navContoller = mockk(relaxed = true)
-        settings = mockk()
-
-        val currentDest: NavDestination = mockk()
-        every { navContoller.currentDestination } returns currentDest
-        every { currentDest.id } returns R.id.crashReporterFragment
+    private val sessionId = "testId"
+    private val components: Components = mockk(relaxed = true)
+    private val settings: Settings = mockk(relaxed = true)
+    private val crash: NativeCodeCrash = mockk(relaxed = true)
+    private var appStore = AppStore(
+        AppState(
+            nonFatalCrashes = listOf(crash)
+        )
+    )
+    private var controller = CrashReporterController(sessionId, components, settings, appStore)
+
+    @Test
+    fun `GIVEN reportCrashes true WHEN handleCloseAndRestore is called THEN try submitting non-fatal crashes and recover tabs`() {
```

Comment:
nitnit :) In general it's better to keep the method names out of the test names. Otherwise all test names will be stale if you anyone ever refactors those methods.

e.g. GIVEN crashes should be reported WHEN user restores tab THEN non-fatal crash report is submitted and tab recovered.

Reply:
(similar for other tests below.)